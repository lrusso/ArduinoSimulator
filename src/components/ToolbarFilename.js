import React from "react"
import { useSimulatorContext } from "../contexts/SimulatorContext"
import { t } from "../utils/service"

const ToolbarFilename = () => {
  const { filename } = useSimulatorContext()
  return (
    <div style={styles.container}>
      <div style={styles.label}>{filename ? filename : t("FILENAME")}</div>
    </div>
  )
}

const styles = {
  container: {
    float: "left",
    display: "flex",
    margin: 0,
    height: "40px",
    alignItems: "center",
  },
  label: {
    float: "left",
    fontFamily: "Arial",
    fontSize: "13px",
    lineHeight: 2.7,
    marginLeft: "6px",
    marginRight: "0px",
    cursor: "default",
    WebkitUserSelect: "none",
    MozUserSelect: "none",
    userSelect: "none",
  },
}

export default ToolbarFilename
