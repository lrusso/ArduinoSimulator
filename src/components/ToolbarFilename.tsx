import React from "react"
import { useSimulatorContext } from "../contexts/SimulatorContext"
import { t } from "../utils/languages"

const ToolbarFilename = () => {
  const { filename } = useSimulatorContext()
  return (
    <div style={styles.container}>
      <div style={styles.label}>{filename ? filename : t("FILENAME")}</div>
    </div>
  )
}

const styles: { [key: string]: React.CSSProperties } = {
  container: {
    float: "left",
    display: "flex",
    margin: 0,
    height: "40px",
    alignItems: "center",
  },
  label: {
    float: "left",
    fontFamily: "'Segoe UI', Tahoma, Geneva, Verdana, sans-serif",
    fontSize: "13px",
    lineHeight: 2.7,
    marginLeft: "6px",
    marginRight: "1rem",
    cursor: "default",
    WebkitUserSelect: "none",
    MozUserSelect: "none",
    userSelect: "none",
  },
}

export default ToolbarFilename
