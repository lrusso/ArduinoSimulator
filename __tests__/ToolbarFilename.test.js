import React from "react"
import renderer from "react-test-renderer"
import ToolbarFilename from "../src/components/ToolbarFilename"

describe("ToolbarFilename", () => {
  test("Rendering 1", () => {
    const tree = renderer.create(<ToolbarFilename />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
