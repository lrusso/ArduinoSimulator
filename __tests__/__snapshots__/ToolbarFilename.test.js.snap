// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolbarFilename Rendering 1 1`] = `
<div
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "float": "left",
      "height": "40px",
      "margin": 0,
    }
  }
>
  <div
    style={
      {
        "MozUserSelect": "none",
        "WebkitUserSelect": "none",
        "cursor": "default",
        "float": "left",
        "fontFamily": "Arial",
        "fontSize": "13px",
        "lineHeight": 2.7,
        "marginLeft": "6px",
        "marginRight": "0px",
        "userSelect": "none",
      }
    }
  >
    Untitled.ino
  </div>
</div>
`;
